app/controllers/queued_videos_controller.rb

```diff
@@ -12,11 +12,43 @@ def create
 
   def destroy
     queued_vid = QueuedVideo.find(params[:id])
-    queued_vid.delete if queued_vid.user == current_user
+    if queued_vid.user == current_user
+      queued_vid.delete
+      current_user.normalize_queued_videos_order
+    end
+    redirect_to queued_videos_path
+  end
+
+  def update
+    begin
+      if params[:queued_videos]
+        update_queued_videos(params[:queued_videos])
+        current_user.normalize_queued_videos_order
+      end
+    rescue ActiveRecord::RecordInvalid
+      flash[:danger] = "Invalid position numbers."
+    rescue
```

Comment:
This rescues all kind of possible bugs you might have. Its better if you can specify what error is the `rescue` rescuing.